addons/mail/static/src/messaging/component/notification_list/notification_list_tests.js

```diff
@@ -0,0 +1,231 @@
+odoo.define('mail.messaging.component.NotificationListTests', function (require) {
+'use strict';
+
+const components = {
+    NotificationList: require('mail.messaging.component.NotificationList'),
+};
+
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('NotificationList', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' }) => {
+            const NotificationListComponent = components.NotificationList;
+            NotificationListComponent.env = this.env;
+            this.component = new NotificationListComponent(null, { filter });
+            await this.component.mount(this.widget.el);
+            await afterNextRender();
+        };
+
+        this.start = async params => {
+            Object.assign(this.data.initMessaging, {
+                channel_slots: {
+                    channel_channel: [{
+                        channel_type: 'channel',
+                        id: 100,
+                        name: "Channel 2019",
+                        message_unread_counter: 0,
+                    }, {
+                        channel_type: 'channel',
+                        id: 200,
+                        name: "Channel 2020",
+                        message_unread_counter: 0,
+                    }],
+                },
+            });
```

Comment:
no data in before each, except definition of fields (which aren't really data though...)